fake/src/impls/geo/mod.rs

```diff
@@ -0,0 +1,94 @@
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Coord::<T> {
+            x: Faker.fake_with_rng(rng),
+            y: Faker.fake_with_rng(rng),
+        }
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Line<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Line::<T> {
+            start: Faker.fake_with_rng(rng),
+            end: Faker.fake_with_rng(rng),
+        }
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::LineString<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::LineString::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiLineString<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiLineString::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Point<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Point::<T>::new(Faker.fake_with_rng(rng), Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiPoint<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiPoint::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Polygon<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::Polygon::<T>::new(Faker.fake_with_rng(rng), Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::MultiPolygon<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        geo_types::MultiPolygon::<T>::new(Faker.fake_with_rng(rng))
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Rect<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        // Rect points can not overlap - this can be optimised to reduce number of uniques required
+        let coords: Vec<T> = crate::unique::<T, _>(rng, 4);
+        let coord_1 = geo_types::Coord::<T> {
+            x: coords[0],
+            y: coords[1],
+        };
+        let coord_2 = geo_types::Coord::<T> {
+            x: coords[2],
+            y: coords[3],
+        };
+        geo_types::Rect::new::<geo_types::Coord<T>>(coord_1, coord_2)
+    }
+}
+
+impl<T: CoordNum + Dummy<Faker>> Dummy<Faker> for geo_types::Triangle<T> {
+    fn dummy_with_rng<R: Rng + ?Sized>(_: &Faker, rng: &mut R) -> Self {
+        // Triangle points can not overlap - this can be optimised to reduce number of uniques required
+        let coords: Vec<T> = crate::unique::<T, _>(rng, 6);
+        let coord_1 = geo_types::Coord::<T> {
+            x: coords[0],
```

Comment:
With triangles we need to make sure we do not have three points in a line, one idea is checking the dx/dy for two combinations of points, and adjusting one point if they match

Reply:
Fixed this.